### ISSUE TITLE
Better names for h3::quic

### ISSUE THREAD

**Issue by eagr** [hyperium/h3#146]
The name `poll_accept_recv` is a bit counter-intuitive. Is it accepting the peer's request for receiving? But it's actually meant for accepting a unidirectional (send) stream from the peer, or in the caller's perspective, *opening* a stream for receiving. So it may not be a good name for the case, as it requires people to do a bit interpretation to figure out what the function is supposed to do.

quinn's API [Connection::accept_uni](https://docs.rs/quinn/latest/quinn/struct.Connection.html#method.accept_uni) feels much more intuitive, accepting unidirectional communication from the peer. And it's clear that it's for receiving in the context.

Proposal for API changes:
* `poll_accept_recv()` -> `poll_accept_uni()`
* `poll_open_send()` -> `poll_open_uni()` (to be symmetric and predictable)
* Maybe?
  * `poll_accept_bidi()` -> `poll_accept_bi()` (as `poll_accept_bi` is more predictable when we have `poll_accept_uni`)
  * `poll_open_bidi()` -> `poll_open_bi()` (to be symmetric)

**Comment by eagr** [comment]
I totally agree that names like `SendStream` and `RecvStream` should stay. I'm just proposing to change some of the function names.

Let me rephrase this a bit. I think the the name `poll_accept_recv` kinda contradicts with what happens. The client initiates a stream (only for sending) to the server and the server accepts it. If we stick to using the word "accept", technically the name should probably be `poll_accept_send(_stream)`. In that sense, putting "accept" and "recv" together may not be a good idea.

Coming back to what we really want to distinguish the functions for
* accept incoming bidirectional stream
* accept incoming unidirectional stream (the direction is quite clear in the context of h3)

These translate directly to `accept_bi` and `accept_uni`. And the combination is more symmetric and intuitive.

**Comment by daalfox** [comment]
I want to start contributing to this library. If you all agree with this change, can I make a PR? @eagr @camshaft @Ruben2424

**Comment by seanmonstar** [comment]
I don't think this issue has an easy answer decided on yet. If you'd like to start contributing (thanks by the way!), a great place to get started is adding more coverage of the duvet comments within the code. You can see the generated report here: https://hyper.rs/h3/ci/compliance/report.html

The reason this is great is because a) you learn how the code works, and b) we improve _keeping track_ of how much of the spec is implemented and tested, and c) it likely involves adding more tests for specific cases.
